ssm/hmm.py

```diff
@@ -1,22 +1,25 @@
-import copy
-import warnings
 from functools import partial
 from tqdm.auto import trange
 
 import autograd.numpy as np
 import autograd.numpy.random as npr
-from autograd.scipy.misc import logsumexp
-from autograd.tracer import getval
-from autograd.misc import flatten
 from autograd import value_and_grad
 
-from ssm.optimizers import adam_step, rmsprop_step, sgd_step, convex_combination
-from ssm.primitives import hmm_normalizer, hmm_expected_states, hmm_filter, hmm_sample, viterbi
-from ssm.util import ensure_args_are_lists, ensure_args_not_none, \
+from .optimizers import adam_step, rmsprop_step, sgd_step, convex_combination
```

Comment:
I'm not a huge fan of relative imports.  I think they're explicitly warned against in PEP8, but I might be wrong.  Anyway, no need to hold up the PR on that account.